src/averageAges.js

```diff
@@ -54,8 +78,35 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = null) {
+  const women = people.filter(person => person.sex === 'f');
+  let children = people.filter(child => people.some(person =>
+    child.mother === person.name));
+
+  if (onlyWithSon) {
+    children = children.filter(child => {
+      return women.find(woman =>
+        woman.name === child.mother) && child.sex === 'm';
+    });
+  }
+
+  const arr = [];
+
+  women.forEach(woman => {
+    const findChildren = children.filter(child =>
+      child.mother === woman.name);
+
+    findChildren.forEach(child => {
+      const ageDiff = child.born - woman.born;
+
+      arr.push(ageDiff);
+    });
+  });
```

Comment:
it is bettter to make with map method.
but better to avoid this iteration and counit ageDiff in reduce below.